tasks/Logger.cpp

```diff
@@ -56,13 +61,38 @@ Logger::~Logger()
 
 bool Logger::startHook()
 {
-    if (_file.value().empty())
-        return false;
+    if(_file.value().empty())
+    {
+      log(Error) << "Could not create log file. Task property _file is empty." << endlog();
+      return false;
+    }
 
-    if(boost::filesystem::exists(_file.value()) && !_overwrite_existing_files.get())
+    if(_overwrite_existing_files.get() && _auto_rename_existing_files.get())
     {
-        log(Error) << "File " << _file.value() << " already exists." << endlog();
-        return false;
+      log(Error) << "Ambiguous property values. Both _overwrite_existing_files and _auto_rename_existing_files are set to true." << endlog();
+      return false;
+    }
+
+    if(boost::filesystem::exists(_file.value()) && !_overwrite_existing_files.get() && _auto_rename_existing_files.get())
+    {
+        log(Warning) << "File " << _file.value() << " already exists." << endlog();
+        // create timestamp
+        time_t now = time(0);
+        tm *t_ptr = localtime(&now);
+        char suffix[21];
+        strftime(suffix, sizeof(suffix), "%F_%H-%M-%S", t_ptr);
+        // append suffix to previous _file.value()
+        vector<string> strs;
+        boost::split(strs, _file.value(), boost::is_any_of("."));
+        strs.insert(strs.end()-1, std::string(suffix));
+        // safety check if timestamped file exists
+        std::string timestamped_str = boost::algorithm::join(strs, ".");
+        if(boost::filesystem::exists(timestamped_str)) {
+          log(Error) << "Timestamped file " << _file.value() << " already exists. Please retry." << endlog();
+          return false;
+        }
+        _file.set(timestamped_str);
```

Comment:
Dynamically changing a property from within the task is really not an expected behavior.

You could create an `attribute` for this (`current_file`). Same general API, just replacing `property` by `attribute` in the orogen file.

Reply:
Ok